flows/waits/base.go

```diff
@@ -0,0 +1,27 @@
+package waits
+
+import (
+	"time"
+
+	"github.com/nyaruka/goflow/flows"
+)
+
+type BaseWait struct {
```

Comment:
So the idea here is that all waits can have a timeout?

Would it make more sense to have a TimeoutWait and have classes that want to have a Timeout as well extend it? That might make it a bit saner in the use of optional fields like Timeout.

PS. Is it TimeOut or Timeout? (test is TimedOut)